lib/iris/cube.py

```diff
@@ -1590,15 +1602,34 @@ def cell_methods(self, cell_methods):
         self._cell_methods = tuple(cell_methods) if cell_methods else tuple()
 
     @property
+    def core_data(self):
+        """
+        The data at the core of this cube.
+        May be a numpy array or a dask array.
+
+        """
+        if self._numpy_array is not None:
+            result = self._numpy_array
+        else:
+            result = self._dask_array
+        return result
```

Comment:
@AlexHilson Is floating the [proposal](https://github.com/SciTools/iris/issues/2394#issuecomment-281656918) that there is benefit to defaulting to `self._dask_array` over `self._numpy_array`.